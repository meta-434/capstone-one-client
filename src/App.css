@import url('https://rsms.me/inter/inter.css');
  html { font-family: 'Inter', sans-serif; }
@supports (font-variation-settings: normal) {
  html { font-family: 'Inter var', sans-serif; }
}
@supports (font-variation-settings: normal) {
  html { font-family: 'Inter Bold', sans-serif; }
}

* {
  box-sizing: border-box;
  font-family: 'Inter';
  background: url('images/background.png') no-repeat center center fixed;
  -webkit-background-size: cover;
  -moz-background-size: cover;
  -o-background-size: cover;
  background-size: cover;
}

body {
  font-size: 20px;
  color: #444;
  text-align: center;
  line-height: 1.5;
}

nav,
footer {
  padding: 1% 0;
}

.error {
  margin-top: 140px;
}

nav {
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  z-index: 10;
  height: 70px;
}

header {
  /*padding: 45px 15px;*/
  margin-top: 70px;
}

section {
  min-height: 200px;
  padding: 30px 15px;
  color: #5E81AC;
}

p {
  margin: 0 auto;
  max-width: 650px;
}

p + p {
  margin-top: 15px;
}

button:disabled {
  border: 2px #4C566A solid;
  border-radius: 3px;
  color: #4C566A;
}

button {
  border: 2px inherit solid;
  border-radius: 3px;
}