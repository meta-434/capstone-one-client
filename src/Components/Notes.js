import React, { Component } from 'react';

class Notes extends Component {
    render() {
        return(
            <div>
                Notes placeholder.
            </div>
        );
    }
}

export default Notes;