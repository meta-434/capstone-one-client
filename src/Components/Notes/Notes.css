header {
  font-size: 40px;
  min-height: 100px;
  padding-top: 20px;
}

.notes-display {
  margin-top: 0;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  flex-wrap: wrap;
}

.notes-display > div {
  margin: 0.5%;
  text-align: center;
  min-width: 200px;
  width: 16.5%;
  max-height: 300px;
  min-height: 300px;
  border: 2px #8da5c4 solid;
  border-radius: 3px;
  flex-direction: column;
}

.notes-display > div > h4,
.notes-display > div > h3 {
  overflow: scroll;
}

.notes-display:first-child {
  border-color: #a3be8c;
}

.notes-display > button {
  border: 2px #8da5c4 solid;
  margin-bottom: 10px;
  color: #8da5c4;
  font-size: 12px;
  text-align: center;
}

.react-form-section {
  padding: 0;
  margin: 0;
}

#submit-button {
  text-align: center;
  color: red;
  margin-right: 50%;
}

@media only screen and (min-width: 575px) {
  .notes-display {
    flex-direction: row;
  }
}
