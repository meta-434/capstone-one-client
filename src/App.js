import React, { Component } from 'react';
import { Route } from 'react-router-dom';
import Nav from './Components/Nav/Nav.js';
import Login from './Components/Login/Login.js';
import Signup from './Components/Signup/Signup.js';
import Main from './Components/Main/Main.js';
import Notes from './Components/Notes/Notes.js';
import Sessions from './Components/Sessions/Sessions.js';
import TimerComponent from './Components/TimerComponent/TimerComponent.js';
import PomodoroContext from './PomodoroContext';
import './App.css'

class App extends Component {
  state = {
    sessions: [],
    notes: [],
    authToken: undefined,
    error: undefined,
  }

  logOut = () => {
    this.setState({authToken: undefined});
    sessionStorage.clear();
  }

  handlePostSignup = ({ username, password }) => {
    fetch(process.env.REACT_APP_SERVER_URL + `/signup`, {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json'
      },
      body: JSON.stringify({
        'username': username,
        'password': password,
      })
    })
        .then(result => {
            let json = result.json() // There's always a response body
            if (result.status >= 200 && result.status < 300) { return json }
            return json.then(Promise.reject.bind(Promise))

        })
        .then(result =>  result.json())
        .catch(error => {
          this.setState({error});
        });
  };

    handlePostAuthenticate = ({ username, password }) => {
        console.log('for login');
        fetch(process.env.REACT_APP_SERVER_URL + `/authenticate`, {
            method: 'post',
            headers: {
                'Content-Type': 'application/json'
            },
            body: JSON.stringify({
                'username': username,
                'password': password,
            })
        })
            .then(result => {
                let json = result.json() // There's always a response body
                if (!(result.status >= 200 && result.status < 300)) {
                    return json.then(Promise.reject.bind(Promise))
                }
                return json;
            })
            .then(resJson => {
                if (!!resJson.token) {
                    this.setState({ authToken: resJson.token, username })
                    sessionStorage.setItem('access-token', this.state.authToken);
                    sessionStorage.setItem('username', username);
                    this.handleGetSessions();
                    this.handleGetNotes();
                }
                else {
                    throw new Error(' error in authenticating. check username and password. ');
                }
            })
            .catch(error => {
                console.error(error);
                this.setState({error});
            });
    };

  handlePostSession = ({session_name, session_description, ownerId}) => {
    fetch(process.env.REACT_APP_SERVER_URL + `/api/sessions`, {
      method: 'post',
      headers: {
        'Content-Type': 'application/json',
        'access-token': `${this.state.authToken || sessionStorage[`access-token`]}`
      },
      body: JSON.stringify({
        session_name,
        session_description,
        session_owner: ownerId
      })
    })
        .then(res => this.handleGetSessions())
        .catch(error => {
          console.error(error);
          this.setState({error});
        });
  }

  handleGetSessions = () => {
    fetch(process.env.REACT_APP_SERVER_URL+ `/api/sessions`, {
      method: 'GET',
      headers: {
        'Content-Type': 'application/json',
        'access-token': `${this.state.authToken || sessionStorage[`access-token`]}`
      }
    })
        .then(response => response.json())
        .then(responseJson => {
              if (responseJson.success && responseJson.success === false) {
                throw new Error('error in getting sessions')
              } else {
                this.setState({
                  sessions: responseJson
                })
              }
            }
        )
        .catch(error => {
          console.error(error);
          this.setState({error});
        });
  }

  handleDeleteSession = (id) => {
    fetch(process.env.REACT_APP_SERVER_URL + `/api/sessions/${id}`, {
      method: 'delete',
      headers: {
        'Content-Type': 'application/json',
        'access-token': `${this.state.authToken || sessionStorage[`access-token`]}`
      }
    })
        .then(res => this.handleGetSessions())
        .catch(error => {
          console.error(error);
          this.setState({error});
        });
  }

  handlePostNote = ({note_name, note_content, ownerId}) => {
    fetch(process.env.REACT_APP_SERVER_URL + `/api/notes`, {
      method: 'post',
      headers: {
        'Content-Type': 'application/json',
        'access-token': `${this.state.authToken || sessionStorage[`access-token`]}`
      },
      body: JSON.stringify({
        note_name,
        note_content,
        note_owner: ownerId
      })
    })
        .then(res => this.handleGetNotes())
        .catch(error => {
          console.error(error);
          this.setState({error});
        });
  }

  handleGetNotes = () => {
    fetch(process.env.REACT_APP_SERVER_URL+ `/api/notes`, {
      method: 'GET',
      headers: {
        'Content-Type': 'application/json',
        'access-token': `${this.state.authToken || sessionStorage[`access-token`]}`
      }
    })
        .then(response => response.json())
        .then(responseJson => {
              if (responseJson.success && responseJson.success === false) {
                throw new Error('error in getting notes')
              } else {
                this.setState({
                  notes: responseJson
                })
              }
            }
        )
        .catch(error => {
          console.error(error);
          this.setState({error});
        });
  }

  handleDeleteNote = (id) => {
    fetch(process.env.REACT_APP_SERVER_URL + `/api/notes/${id}`, {
      method: 'delete',
      headers: {
        'Content-Type': 'application/json',
        'access-token': `${this.state.authToken || sessionStorage[`access-token`]}`
      }
    })
        .then(res => this.handleGetNotes())
        .catch(error => {
          console.error(error);
          this.setState({error});
        });
  }

  render () {
    const { sessions, notes, authToken, username, error } = this.state;
    const context = {
      sessions,
      notes,
      username,
      authToken,
      error,
      handlePostAuthenticate: this.handlePostAuthenticate,
      handlePostSignup: this.handlePostSignup,
      logOut: this.logOut,
      handleGetNotes: this.handleGetNotes,
      handlePostNote: this.handlePostNote,
      handleDeleteNote: this.handleDeleteNote,
      handleGetSessions: this.handleGetSessions,
      handleDeleteSession: this.handleDeleteSession,
      handlePostSession: this.handlePostSession,
    }
    return (
        <PomodoroContext.Provider value={(context)}>
          <div className="app">
            <Route path="/" render={routeProps => <Nav {...routeProps}/>}/>
            <main className="app-content">
              <Route exact path="/" component={Main} />
              <Route path="/login" component={Login}/>
              <Route path="/signup" component={Signup} />
              <Route path="/timer" component={TimerComponent}/>
              <Route path="/sessions/:sessionId?" component={Sessions}/>
              <Route path="/notes/:noteId?" component={Notes}/>

            </main>
          </div>
        </PomodoroContext.Provider>
    );
  }
}

export default App;