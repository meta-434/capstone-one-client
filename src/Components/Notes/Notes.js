import React, { Component } from 'react';
import PomodoroContext from "../../PomodoroContext";
import './Notes.css'
import ErrorDisplay from "../ErrorDisplay/ErrorDisplay";

class Notes extends Component {

    static contextType = PomodoroContext;

    constructor() {
        super();

        this.state = {
            name: undefined,
            content: undefined,
            nameValid: false,
            contentValid: false,
        }
    }

    componentDidMount() {
        this.context.handleGetNotes();
        this.validateName(this.state.name);
        this.validateContent(this.state.content);
    }

    handleDelete = (id) => {
        this.context.handleDeleteNote(id);
    }

    handlePostSubmit = (e) => {
        e.preventDefault();
        const {name, content} = this.state;
        this.context.handlePostNote({note_name: name, note_content: content});
        // this.context.handleGetNotes();
    }

    handleNoteName = (e) => {
        let nameInput = e.target.value;
        this.setState({ name: nameInput }, () => this.validateName(nameInput));
    }

    handleNoteContent = (e) => {
        let contentInput = e.target.value;
        this.setState({ content: contentInput }, () => this.validateContent(contentInput))
    };

    validateName(name) {
        let validationMessages;
        let hasError = false;

        if (!name) {
            hasError = true;
            validationMessages = ' name cannot be blank. '
        }
        else {
            validationMessages = '';
        }

        this.setState({
            nameValid: !hasError,
            nameValidation: validationMessages,
        }, () =>this.nameValid(name));
    };

    validateContent(content) {
        let validationMessages;
        let hasError = false;

        if (!content) {
            hasError = true;
            validationMessages = ' content cannot be blank. '
        }

        else {
            validationMessages = '';
        }

        this.setState({
            contentValid: !hasError,
            contentValidation: validationMessages,
        }, () => this.contentValid(content));
    };

    nameValid(name) {
        if (this.state.name) {
            this.setState({name: name})
        }
    };

    contentValid(content) {
        if (this.state.content) {
            this.setState({content})
        }
    };

    render() {
        return(
            <>
                {
                    (!!this.context.error)
                        ? <ErrorDisplay />
                        : ''
                }
                <section>
                    <header>My Notes</header>
                </section>
                <section className="react-form-section">
                    <form
                        className="react-form"
                        onSubmit={this.handlePostSubmit}>
                        <label htmlFor="note-name">note name: </label>
                        <input
                            type="text"
                            id="note-name"
                            name="note-name"
                            className="note-name"
                            onChange={this.handleNoteName}
                            defaultValue={'enter note name'}
                            aria-label="note name"
                            aria-required="true"
                            aria-describedby="error-box"
                        />
                        <label htmlFor="note-content">note content: </label>
                        <textarea
                            id="note-content"
                            name="note-content"
                            className="note-content"
                            onChange={this.handleNoteContent}
                            defaultValue={'enter note content'}
                            aria-label="note content"
                            aria-required="true"
                            aria-describedby="error-box"
                        />
                        <button
                            className="submit-button"
                            type="submit"
                            disabled={!this.state.nameValid || !this.state.contentValid}>
                            Submit
                        </button>
                    </form>
                    <section className="error-box" id="error-box" aria-live="assertive">
                        {this.state.nameValidation}
                        <br />
                        {this.state.contentValidation}
                    </section>
                </section>
                <section className="notes-display">
                    <PomodoroContext.Consumer>
                        {({notes}) => {
                            if (notes.length !== 0) {
                                return notes.map((note, index) => {
                                    return (
                                        <div className="notes-display" key={index} id={note.id}>
                                            <h3>{note.note_name}</h3>
                                            <h4>â†³ {note.note_content}</h4>
                                            <button onClick={() => this.handleDelete(note.id)}>Delete</button>
                                        </div>
                                    );
                                }).reverse();
                            }
                        }}
                    </PomodoroContext.Consumer>
                </section>
            </>
        );
    }
}

export default Notes;