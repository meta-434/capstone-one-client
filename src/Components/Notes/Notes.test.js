import React from 'react';
import { render } from '@testing-library/react';
import Notes from './Notes';
import App from '../../App';
import {BrowserRouter} from "react-router-dom";
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import PomodoroContext from "../../PomodoroContext";

test('renders Notes component without crashing', () => {

    const div = document.createElement('div');
    ReactDOM.render(
            <PomodoroContext.Provider value={{
                notes: [
                    {
                        note_id: 1,
                        note_name: 'jest note',
                        note_content: 'jest test note',
                        note_owner: 30,
                    },
                ],
                handleGetNotes: () => {
                    return {
                        note_name: 'jest note',
                        note_content: 'jest note contents',
                        note_owner: 1
                    };
                }
            }}>
                <Notes />
            </PomodoroContext.Provider>
        , div);
    ReactDOM.unmountComponentAtNode(div);
});

it('renders the Notes UI as expected', () => {
    const tree = renderer
        .create(
            <PomodoroContext.Provider value={{
                notes: [
                    {
                        note_id: 1,
                        note_name: 'jest note',
                        note_content: 'jest test note',
                        note_owner: 30,
                    },
                ],
                handleGetNotes: () => {
                    return {
                        note_name: 'jest note',
                        note_content: 'jest note contents',
                        note_owner: 1
                    };
                }
            }}>
                <Notes />
            </PomodoroContext.Provider>

        )
        .toJSON();

    expect(tree).toMatchSnapshot();
});

it('adds notes and displays them after submission', () => {
    const newNote = {note_name: 'jest note submission', note_content: 'jest note submission'};

    const component = renderer.create(
        <PomodoroContext.Provider value={{
            notes: [
                {
                    note_id: 1,
                    note_name: 'jest note',
                    note_content: 'jest test note',
                    note_owner: 30,
                },
            ],
            handleGetNotes: () => {
                return {
                    note_name: 'jest note',
                    note_content: 'jest note contents',
                    note_owner: 1
                };
            }
        }}>
            <Notes />
        </PomodoroContext.Provider>
    );

    //const instance = component.getInstance();

    let tree = component.toJSON();
    // pre-submission


    expect(tree).toMatchSnapshot();

})