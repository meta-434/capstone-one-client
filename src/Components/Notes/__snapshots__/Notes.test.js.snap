// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`adds notes and displays them after submission 1`] = `
Array [
  "",
  <section>
    <header>
      My Notes
    </header>
  </section>,
  <section
    className="react-form-section"
  >
    <form
      className="react-form"
      onSubmit={[Function]}
    >
      <label
        htmlFor="note-name"
      >
        note name: 
      </label>
      <input
        aria-describedby="error-box"
        aria-label="note name"
        aria-required="true"
        className="note-name"
        id="note-name"
        name="note-name"
        onChange={[Function]}
        placeholder="enter note name"
        type="text"
      />
      <label
        htmlFor="note-content"
      >
        note content: 
      </label>
      <textarea
        aria-describedby="error-box"
        aria-label="note content"
        aria-required="true"
        className="note-content"
        id="note-content"
        name="note-content"
        onChange={[Function]}
        placeholder="enter note content"
      />
      <button
        className="submit-button"
        disabled={true}
        type="submit"
      >
        Submit
      </button>
    </form>
    <section
      aria-live="assertive"
      className="error-box"
      id="error-box"
    >
       name cannot be blank. 
      <br />
       content cannot be blank. 
    </section>
  </section>,
  <section
    className="notes-display"
  >
    <div
      className="notes-display"
    >
      <h3
        data-testid="name"
      >
        jest note
      </h3>
      <h4
        data-testid="content"
      >
        ↳ 
        jest test note
      </h4>
      <button
        data-testid="delbutton"
        onClick={[Function]}
      >
        Delete
      </button>
    </div>
  </section>,
]
`;

exports[`renders the Notes UI as expected 1`] = `
Array [
  "",
  <section>
    <header>
      My Notes
    </header>
  </section>,
  <section
    className="react-form-section"
  >
    <form
      className="react-form"
      onSubmit={[Function]}
    >
      <label
        htmlFor="note-name"
      >
        note name: 
      </label>
      <input
        aria-describedby="error-box"
        aria-label="note name"
        aria-required="true"
        className="note-name"
        id="note-name"
        name="note-name"
        onChange={[Function]}
        placeholder="enter note name"
        type="text"
      />
      <label
        htmlFor="note-content"
      >
        note content: 
      </label>
      <textarea
        aria-describedby="error-box"
        aria-label="note content"
        aria-required="true"
        className="note-content"
        id="note-content"
        name="note-content"
        onChange={[Function]}
        placeholder="enter note content"
      />
      <button
        className="submit-button"
        disabled={true}
        type="submit"
      >
        Submit
      </button>
    </form>
    <section
      aria-live="assertive"
      className="error-box"
      id="error-box"
    >
       name cannot be blank. 
      <br />
       content cannot be blank. 
    </section>
  </section>,
  <section
    className="notes-display"
  >
    <div
      className="notes-display"
    >
      <h3
        data-testid="name"
      >
        jest note
      </h3>
      <h4
        data-testid="content"
      >
        ↳ 
        jest test note
      </h4>
      <button
        data-testid="delbutton"
        onClick={[Function]}
      >
        Delete
      </button>
    </div>
  </section>,
]
`;
